examples/silentpayments.c

```diff
@@ -0,0 +1,503 @@
+                 *  ignore these steps and simply pass `NULL` for the
+                 *  label_lookup and label_context arguments:
+                 *
+                 *      _silentpayments_recipient_scan_outputs(..., NULL, NULL);
+                 *
+                 *  In this case, since Bob has access to the full transaction
+                 *  outputs when scanning its easy for him to scan with labels,
+                 *  as demonstrated below. For efficient scanning, Bob keeps a
+                 *  cache of every label he has previously used and uses a
+                 *  callback to check if a potential label exists in his cache.
+                 *  Since the labels are created using an incremental integer
+                 *  `m`, if Bob ever forgets how many labels he has previously
+                 *  used, he can pregenerate a large number of labels e.g.,
+                 *  0..100_000.
+                 */
+                size_t len = 33;
+                secp256k1_pubkey label;
+                unsigned int m = 1;
+
+                /* Load Bob's spend public key */
+                ret = secp256k1_ec_pubkey_parse(ctx,
+                    &spend_pubkey,
+                    bob_spend_pubkey,
+                    33
+                );
+
+                /* Add an entry to the cache. This implies Bob has previously
+                 * called
+                 * `secp256k1_silentpayments_recipient_create_labelled_spend_pubkey`
+                 * and is using the resulting labelled spend pubkey to encode a
+                 * labelled silent payments address.
+                 */
+                ret = secp256k1_silentpayments_recipient_create_label_tweak(ctx,
+                    &label,
+                    labels_cache.entries[0].label_tweak,
+                    bob_scan_key,
+                    m
+                );
+                secp256k1_ec_pubkey_serialize(ctx,
+                    labels_cache.entries[0].label,
+                    &len,
+                    &label,
+                    SECP256K1_EC_COMPRESSED
+                );
+                assert(ret);
+                labels_cache.entries_used = 1;
+            }
+
+            /* Bob collects the public data from the transaction inputs and
+             * creates a `secp256k1_silentpayments_public_data` object. He uses
+             * this for his own scanning and also serializes the `public_data`
+             * object to send to light clients. We will use this later for
+             * Carol, who is scanning as a light client. Note, anyone can create
+             * and provide these `public_data` objecs, i.e. you don't need to be
+             * a silent payments wallet, just someone interested in vending this
+             * data to light clients, e.g. a wallet service provider. In our
+             * example, Bob is scanning for himself but also sharing this data
+             * with light clients.
+             */
```

Comment:
```suggestion
            /* Bob collects the public data from the transaction inputs and
             * creates a `secp256k1_silentpayments_public_data` object. He uses
             * this for his own scanning and also serializes the `public_data`
             * object to send to light clients. We will use this later for
             * Carol, who is scanning as a light client. Note, anyone can create
             * and provide these `public_data` objects, i.e. you don't need to be
             * a silent payments wallet, just someone interested in vending this
             * data to light clients, e.g. a wallet service provider. In our
             * example, Bob is scanning for himself but also sharing this data
             * with light clients.
             */
```

nit: Also, "vending" sounds a bit strange.